tests/by-util/test_date.rs

```diff
@@ -259,6 +285,42 @@ fn test_date_set_permissions_error() {
     }
 }
 
+#[test]
+#[cfg(all(unix, not(any(target_os = "android", target_os = "macos"))))]
+fn test_date_set_permissions_error_interpreted() {
+    // This implicitly tests that the given strings are interpreted as valid dates,
+    // because parsing errors would have been discovered earlier in the process.
+    if !(geteuid() == 0 || uucore::os::is_wsl_1()) {
+        for date_string in [
+            "yesterday",
+            // TODO "a fortnight ago",
+            "42 days",
+            "2001-02-03",
+            "20010203",
+            // TODO "02/03/2001",
+        ] {
+            let result = new_ucmd!().arg("-s").arg(date_string).fails();
+            // stdout depends on the specific date; don't check it.
```

Comment:
Here it might make sense to test that stdout is not empty because currently there is a bug in our implementation: we don't output anything to stdout whereas GNU `date` outputs the specified date.
```
$ cargo run -q date --set "yesterday"
date: cannot set date: Permission denied
$ date --set "yesterday"
date: cannot set date: Operation not permitted
Tue Feb 18 05:06:33 PM CET 2025
```
```suggestion
            // stdout depends on the specific date; don't check it.
            assert!(!result.stdout_str().is_empty());
```